src/test/java/bio/terra/service/dataset/DatasetBucketDaoTest.java

```diff
@@ -138,22 +141,37 @@ public void TestGetProjectForDatasetProfileCombo() throws Exception {
         "Should NOT retrieve existing project",
         datasetBucketDao.getProjectResourceForBucket(datasetId, billingProfile2.getId()));
 
+    List<UUID> projectResourceIds =
+        datasetBucketDao.getProjectResourceIdsForBucketPerDataset(datasetId);
+    assertEquals("Just one bucket linked right now", 1, projectResourceIds.size());
+
+    // Link dataset to new billing profile via bucket
+    GoogleProjectResource ingestProjectResource =
+        ResourceFixtures.randomProjectResource(billingProfile2);
+    UUID ingestProjectId = resourceDao.createProject(ingestProjectResource);
+    ingestProjectResource.id(ingestProjectId);
+    UUID ingestBucketResourceId = createBucketDbEntry(ingestProjectResource);
+    datasetBucketDao.createDatasetBucketLink(datasetId, ingestBucketResourceId);
+
+    List<UUID> projectResourceIds_after =
+        datasetBucketDao.getProjectResourceIdsForBucketPerDataset(datasetId);
+    assertEquals(
+        "Should return both projects for both billing profiles for dataset 1",
+        2,
+        projectResourceIds_after.size());
```

Comment:
[Code review note]: This is the chunk of code that test my dao changes. I'm adding another link between a dataset and bucket, but with a different billing profile and proving that that increases the count by 1.